### ISSUE TITLE
JavaScript Required

### ISSUE THREAD

**Issue by rwxrob** [artifacthub/hub#1284]
When I go to the main site with Lynx or W3M or with my JavaScript disabled in my graphic web browser I'm not even allowed to read the page because "This app requires JavaScript." No modern website should discount the importance of "progressive enhancement" in web design. Requiring that JavaScript be enabled just to read the main page is a bug at the fundamental level of accessibility as well. There is no solid rational for making the readable elements of this site -- including the application components -- anything but prerendered HTML and CSS. Would you create all of this with Flash? If not, then why force the same unnecessary requirement on everyone.

**Comment by tegioz** [comment]
Hi @rwxrob

I'll start with the sentence below that you edited but we still received by email:

> Currently, this site and application is tantamount to a big, buggy Flash app.

We really appreciate your feedback, but we encourage you to give it a more respectful way. Comments like this are as unhelpful as unneeded, and IMHO they tend to hurt the Open Source ecosystem.

Said that, I will respectfully try to answer to your points.

> No modern website should discount the importance of "progressive enhancement" in web design.

As of today, depending on the source you check, you'll find out that the percentage of users with Javascript disabled ranges from 0.0X% to 0.000X%. Projects usually have limited resources, which means that decisions needs to be made about what should and shouldn't be done. Ideally any website should work on any device, on any browser. But in practice this usually isn't the case, even for top sites, as making it happen is tricky and consumes a lot of time. Trying to cover that 0.001% may have a negative impact on the features you can provide to the remaining 99.999% of users. It's a trade-off, and I understand that not everyone will agree with our choices, specially if they are in the segment left out (by choice or by need).

> Would you create all of this with Flash?

If it was a vendor neutral, safe, standardized and mature technology supported in all modern browsers out of the box that gave me an important advantage, I most definitely would 🙂 

Regarding the accessibility point, we thank you for bringing that up, as it's a topic we haven't paid the attention it deserves yet. So we'll work on it soon to improve that aspect, even though Javascript will still be required.

**Comment by tegioz** [comment]
We've just deployed [a lot of changes](https://github.com/artifacthub/hub/commit/87a1fedb4239c2b394d6a522c404791c3b75de2d) to improve the web application's accessibility. Lighthouse accessibility checks results are actually pretty good now (see screenshot below). We've also tested the site using a screen reader and the experience has improved dramatically as well.

<img width="1129" alt="accesibility" src="https://user-images.githubusercontent.com/1213902/118112921-ecffc580-b3e5-11eb-8afc-e8600d21419f.png">

Closing this ticket as there isn't anything else pending we have plans to address for now.
